rest/route/host_spawn.go

```diff
@@ -1013,6 +1013,66 @@ func (h *getVolumesHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(volumeDocs)
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// GET /rest/v2/volumes
+
+type getVolumeByIDHandler struct {
+	volumeID string
+	sc       data.Connector
+}
+
+func makeGetVolumeByID(sc data.Connector) gimlet.RouteHandler {
+	return &getVolumeByIDHandler{
+		sc: sc,
+	}
+}
+
+func (h *getVolumeByIDHandler) Factory() gimlet.RouteHandler {
+	return &getVolumeByIDHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *getVolumeByIDHandler) Parse(ctx context.Context, r *http.Request) error {
+	var err error
+	if h.volumeID, err = validateID(gimlet.GetVars(r)["volume_id"]); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *getVolumeByIDHandler) Run(ctx context.Context) gimlet.Responder {
+	v, err := h.sc.FindVolumeById(h.volumeID)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(err)
+	}
+	if v == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    "volume not found",
+		})
+	}
+	volumeDoc := &model.APIVolume{}
+	if err = volumeDoc.BuildFromService(v); err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "err converting volume '%s' to API model", v.ID))
+	}
+	// if the volume is attached to a host, also return the host ID and volume device name
+	attachedHost, err := h.sc.FindHostWithVolume(v.ID)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "error querying for host"))
+	}
+	if attachedHost != nil {
+		volumeDoc.HostID = model.ToStringPtr(attachedHost.Id)
```

Comment:
why do we need to replace the volume document's host field?

Reply:
🤷 I can remove it, I copied the other get volumes route but I think it's a holdover from before volumes had a host field.